Emby.Server.Implementations/Data/BaseSqliteRepository.cs

```diff
@@ -101,63 +110,115 @@ protected BaseSqliteRepository(ILogger<BaseSqliteRepository> logger)
+            ReadConnections = new ConnectionPool(ReadConnectionsCount, CreateReadConnection);
+        }
 
         protected ManagedConnection GetConnection(bool readOnly = false)
         {
-            WriteLock.Wait();
-            if (WriteConnection is not null)
+            if (readOnly)
             {
-                return new ManagedConnection(WriteConnection, WriteLock);
+                return ReadConnections.GetConnection();
             }
 
-            WriteConnection = SQLite3.Open(
+            return WriteConnections.GetConnection();
+        }
+
+        protected SQLiteDatabaseConnection CreateWriteConnection()
+        {
+            var writeConnection = SQLite3.Open(
                 DbFilePath,
                 DefaultConnectionFlags | ConnectionFlags.Create | ConnectionFlags.ReadWrite,
                 null);
 
             if (CacheSize.HasValue)
             {
-                WriteConnection.Execute("PRAGMA cache_size=" + CacheSize.Value);
+                writeConnection.Execute("PRAGMA cache_size=" + CacheSize.Value);
             }
 
             if (!string.IsNullOrWhiteSpace(LockingMode))
             {
-                WriteConnection.Execute("PRAGMA locking_mode=" + LockingMode);
+                writeConnection.Execute("PRAGMA locking_mode=" + LockingMode);
             }
 
             if (!string.IsNullOrWhiteSpace(JournalMode))
             {
-                WriteConnection.Execute("PRAGMA journal_mode=" + JournalMode);
+                writeConnection.Execute("PRAGMA journal_mode=" + JournalMode);
             }
 
             if (JournalSizeLimit.HasValue)
             {
-                WriteConnection.Execute("PRAGMA journal_size_limit=" + JournalSizeLimit.Value);
+                writeConnection.Execute("PRAGMA journal_size_limit=" + JournalSizeLimit.Value);
             }
 
             if (Synchronous.HasValue)
             {
-                WriteConnection.Execute("PRAGMA synchronous=" + (int)Synchronous.Value);
+                writeConnection.Execute("PRAGMA synchronous=" + (int)Synchronous.Value);
             }
 
             if (PageSize.HasValue)
             {
-                WriteConnection.Execute("PRAGMA page_size=" + PageSize.Value);
+                writeConnection.Execute("PRAGMA page_size=" + PageSize.Value);
             }
 
-            WriteConnection.Execute("PRAGMA temp_store=" + (int)TempStore);
+            writeConnection.Execute("PRAGMA temp_store=" + (int)TempStore);
 
             // Configuration and pragmas can affect VACUUM so it needs to be last.
-            WriteConnection.Execute("VACUUM");
+            writeConnection.Execute("VACUUM");
 
-            return new ManagedConnection(WriteConnection, WriteLock);
+            return writeConnection;
+        }
+
+        protected SQLiteDatabaseConnection CreateReadConnection()
+        {
+            var writeConnection = SQLite3.Open(
```

Comment:
```suggestion
            var readConnection = SQLite3.Open(
```

Just a minor inconsistency with the naming.